### DISCUSSION TITLE
Nullable<T>.GetValueOrDefault() should be implicit in a nullable context

### DISCUSSION THREAD

**Discussion by marinasundstrom** [dotnet/csharplang#4197] [Language Ideas] [answered]
One thing that bugs me:

Currently you have to call  ```GetValueOrDefault()``` to cast a Nullable<T> regardless whether ```nullable``` is enabled or not:

```
int? value = null;

if(value != null)
{
    int foo = value.GetValueOrDefault();
}
```

In a Nullable context, the ```GetValueOrDefault()```call (the type cast) should be implicit (emitted by the compiler) when checked, just like with nullable reference types:

```
int? value = 42;

if(value != null)
{
    int foo = value;
}
```

This ensures consistency. I don't think that a change would break anything existing in a major way.

**Comment by YairHalberstadt** [comment, marked as answer]
Discussed here: https://github.com/dotnet/csharplang/issues/1865. Difficult to be sure, but I think https://github.com/dotnet/csharplang/blob/master/meetings/2018/LDM-2018-10-24.md#using-nullable-values-as-non-nullable-values implies they're not going to do this.
